src/grt/src/fastroute/src/maze3D.cpp

```diff
@@ -1423,13 +1429,13 @@ void FastRouteCore::mazeRouteMSMDOrder3D(int expand,
           if (grids[i].x == grids[i + 1].x)  // a vertical edge
           {
             const int min_y = std::min(grids[i].y, grids[i + 1].y);
-            graph2d_.addUsageV(grids[i].x, min_y, net->getEdgeCost());
+            graph2d_.updateUsageV(grids[i].x, min_y, net, net->getEdgeCost());
             v_edges_3D_[grids[i].layer][min_y][grids[i].x].usage
                 += net->getLayerEdgeCost(grids[i].layer);
-          } else  /// if(gridsY[i]==gridsY[i+1])// a horizontal edge
+          } else  /// if(grids[i].y==grids[i+1].y)// a horizontal edge
```

Comment:
Not really related to your PR, but you can remove this comment. Maybe just leave `a horizontal edge`, no need for the whole check.